tests/test_representations.py

```diff
@@ -1,2 +1,293 @@
+    ReplicationTarget,
+    RepresentationLog,
+)
+from fractal_database.representations import Representation
+from fractal_database_matrix.models import MatrixCredentials, MatrixReplicationTarget
+from fractal_database_matrix.representations import (
+    MatrixExistingSubSpace,
+    MatrixRepresentation,
+    MatrixRoom,
+    MatrixSpace,
+    MatrixSubRoom,
+    MatrixSubSpace,
+)
+from nio import RoomGetStateEventResponse, SpaceGetHierarchyResponse
+
+pytestmark = pytest.mark.django_db(transaction=True)
+
+
+async def test_put_state_no_creds():
+    matrix_representation = MatrixRepresentation()
+
+    room_id = "room_id"
+    target = MagicMock()
+    state_type = "state_type"
+    content = {}
+
+    with patch(
+        "fractal.cli.controllers.auth.AuthenticatedController.get_creds", return_value=None
+    ):
+        with pytest.raises(Exception) as e:
+            await matrix_representation.put_state(room_id, target, state_type, content)
+        assert str(e.value) == "You must be logged in to put state"
+
+
+async def test_create_room_no_creds():
+    matrix_representation = MatrixRepresentation()
+
+    target = MagicMock()
+    name = "Test Room"
+    initial_state = []
+    public = False
+    invite = []
+
+    with patch(
+        "fractal.cli.controllers.auth.AuthenticatedController.get_creds", return_value=None
+    ):
+        with pytest.raises(Exception) as e:
+            await matrix_representation.create_room(
+                target, name, initial_state=initial_state, public=public, invite=invite
+            )
+        assert str(e.value) == "You must be logged in to create a room"
+
+
+async def test_create_room_invalid_invite_uppercase(
+    logged_in_db_auth_controller: AuthenticatedController,
+):
+    matrix_representation = MatrixRepresentation()
+    creds = AuthenticatedController().get_creds()
+
+    target = MagicMock()
+    name = "Test Room"
+    initial_state = []
+    public = False
+    # Simulate uppercase Matrix IDs in the invite list
+    invite = ["@Test:Upper.com"]
+
+    with patch(
+        "fractal.cli.controllers.auth.AuthenticatedController.get_creds", return_value=creds
+    ):
+        with pytest.raises(Exception) as e:
+            await matrix_representation.create_room(
+                target, name, initial_state=initial_state, public=public, invite=invite
+            )
+        assert str(e.value) == "Matrix IDs must be lowercase"
+
+
+# Broken with Update
+"""
+async def test_create_room_representation_success():
+    matrix_representation = MatrixRepresentation()
+    target = MagicMock()
+    room_id = "test room"
+
+    with patch(
+        "fractal.cli.controllers.auth.AuthenticatedController.get_creds",
+        return_value=("access_token", "homeserver_url", "matrix_id"),
+    ), patch(
+        "fractal.matrix.async_client.FractalAsyncClient.room_create",
+        return_value=MagicMock(room_id="room_id"),
+    ), patch(
+        "builtins.print"
+    ) as mocked_print:
```

Comment:
Let's avoid mocking print. There should be some way to ensure that the function you called did something. In this case, it should have called room_create